src/test/java/org/broadinstitute/hellbender/engine/ReferenceFileSourceUnitTest.java

```diff
@@ -12,7 +16,61 @@
 
     @Test(expectedExceptions = UserException.MissingReference.class)
     public void testMissingReferenceFile() throws IOException {
-        new ReferenceFileSource(GATKBaseTest.getSafeNonExistentFile("NonExistentReference.fasta").getAbsolutePath());
+        new org.broadinstitute.hellbender.engine.datasources.ReferenceFileSource(
+                GATKBaseTest.getSafeNonExistentFile("NonExistentReference.fasta")
+                        .getAbsolutePath());
+    }
+
+    @Test(expectedExceptions = UserException.MissingReferenceFaiFile.class)
+    public void testReferenceInPathWithoutFai() throws IOException {
+        try (FileSystem jimfs = Jimfs.newFileSystem(Configuration.unix())) {
+            final Path refPath = jimfs.getPath("reference.fasta");
+            Files.createFile(refPath);
+            final Path dictPath = jimfs.getPath("reference.dict");
+            Files.createFile(dictPath);
+
+            new org.broadinstitute.hellbender.engine.ReferenceFileSource(refPath);
+        }
+    }
+
+    @Test(expectedExceptions = UserException.MissingReferenceDictFile.class)
+    public void testReferenceInPathWithoutDict() throws IOException {
+        try (FileSystem jimfs = Jimfs.newFileSystem(Configuration.unix())) {
+            final Path refPath = jimfs.getPath("reference.fasta");
+            Files.createFile(refPath);
+            final Path faiFile = jimfs.getPath("reference.fasta.fai");
+            Files.createFile(faiFile);
+
+            new org.broadinstitute.hellbender.engine.ReferenceFileSource(refPath);
+        }
+    }
+
+    @Test
+    public void testReferenceInPath() throws IOException {
+        try (FileSystem jimfs = Jimfs.newFileSystem(Configuration.unix())) {
+            final Path refPath = jimfs.getPath("reference.fasta");
+            Files.createFile(refPath);
+            final Path faiFile = jimfs.getPath("reference.fasta.fai");
+            Files.createFile(faiFile);
+            final Path dictPath = jimfs.getPath("reference.dict");
+            Files.createFile(dictPath);
+
+            new org.broadinstitute.hellbender.engine.ReferenceFileSource(refPath);
+        }
+    }
+
+    @Test
+    public void testDatasourcesReferenceInPath() throws IOException {
+        try (FileSystem jimfs = Jimfs.newFileSystem(Configuration.unix())) {
+            final Path refPath = jimfs.getPath("reference.fasta");
+            Files.createFile(refPath);
+            final Path faiFile = jimfs.getPath("reference.fasta.fai");
+            Files.createFile(faiFile);
+            final Path dictPath = jimfs.getPath("reference.dict");
+            Files.createFile(dictPath);
+
+            new org.broadinstitute.hellbender.engine.datasources.ReferenceFileSource(refPath);
```

Comment:
This is so confusing... We should fix this naming collision..  I opened a ticket for it #3926.

I think this test should probably be moved to a new test class ReferenceFileSourceUnitTest in the same package as that.  If it could assert that we can really read data as well that would be great.

Reply:
Agreed; I split the class in two accordingly. Remains to actually read the data.